pkg/pillar/hypervisor/containerd.go

```diff
@@ -115,25 +115,37 @@ func (ctx ctrdContext) Task(status *types.DomainStatus) types.Task {
 }
 
 func (ctx ctrdContext) setupSpec(status *types.DomainStatus, config *types.DomainConfig,
-	aa *types.AssignableAdapters, globalConfig *types.ConfigItemValueMap,
 	volume string) (containerd.OCISpec, error) {
 	spec, err := ctx.ctrdClient.NewOciSpec(status.DomainName, config.Service)
 	if err != nil {
 		logError("failed to create OCI spec for domain %s: %v", status.DomainName, err)
 		return nil, err
 	}
-	ociSpec := spec.Get()
 
+	if err := spec.UpdateFromVolume(volume); err != nil {
```

Comment:
@christoph-zededa , are you changing the order of the CDI processing? CDI injection must come before these OCI update functions.... but let me test first to see if it works as it is...

Reply:
While implementing native container CDI support I remember that if I do the CDI processing after these updates, it was overwriting them, but my branch was at early stages, I'm testing to verify it doesn't happen the same now, but it takes time to deploy a 4GB CUDA container on a developer kit with SD Card, I will post the results here....

Reply:
It works as it is. You don't need to change.